core/src/main/java/com/datastax/oss/driver/api/core/config/OptionsMap.java

```diff
@@ -255,7 +255,7 @@ protected static void fillWithDriverDefaults(OptionsMap map) {
     map.put(TypedDriverOption.LOAD_BALANCING_POLICY_SLOW_AVOIDANCE, true);
     map.put(TypedDriverOption.SESSION_LEAK_THRESHOLD, 4);
     map.put(TypedDriverOption.CONNECTION_CONNECT_TIMEOUT, Duration.ofSeconds(5));
-    map.put(TypedDriverOption.CONNECTION_INIT_QUERY_TIMEOUT, Duration.ofMillis(500));
+    map.put(TypedDriverOption.CONNECTION_INIT_QUERY_TIMEOUT, Duration.ofSeconds(5));
```

Comment:
`reference.conf` uses HOCON substitutions to define a few other options in terms of this one. You need to update them too: `CONNECTION_SET_KEYSPACE_TIMEOUT`, `HEARTBEAT_TIMEOUT`, `CONTROL_CONNECTION_TIMEOUT` and `REPREPARE_TIMEOUT`.

In fact we should probably extract a variable:
```java
    Duration initQueryTimeout = Duration.ofSeconds(5);
    map.put(TypedDriverOption.CONNECTION_INIT_QUERY_TIMEOUT, initQueryTimeout);
    map.put(TypedDriverOption.CONNECTION_SET_KEYSPACE_TIMEOUT, initQueryTimeout);
    ...
```

Can you also do a pass to handle all other substitutions the same way, and commit that separately?

`MapBasedDriverConfigLoaderTest.should_fill_default_profile_like_reference_file` will let you know if everything is in sync.

Reply:
done, what do you mean in a separate commit? Do you want to have two commits in the `4.x` after this branch will be merged?

Reply:
Yes, because the refactoring to handle other substitutions by extracting a local variable is unrelated to this PR.